### PR TITLE
Add Course and related classes to DG

### PR DESCRIPTION
**Pull request by maximustan2**

### CODE REVIEW

**Review by accountexeregister — APPROVED** [review]
LGTM

File: docs/diagrams/CourseClassDiagram.puml
Comment (FlyingDonkeys) [inline docs/diagrams/CourseClassDiagram.puml:1]: Good to have a course class diagram to clarify our Course model with other developers

File: docs/UserGuide.md [hunk 1]
```diff
@@ -148,6 +148,20 @@ Examples:
 * `list` followed by `delete 2` deletes the 2nd person in the address book.
 * `find Betsy` followed by `delete 1` deletes the 1st person in the results of the `find` command.
 
+### Adding a course: `course-add`
+
+Adds a course to TAHub Contacts.
```
Comment (FlyingDonkeys) [inline docs/UserGuide.md:153]: Not sure if the edit, delete methods etc should also be added into the User Guide?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -148,6 +148,20 @@ Examples:
...
+
+Format: `course-add c/COURSE_CODE n/COURSE_NAME`
+
+Examples:
+* course-add c/CS1101S n/Programming Methodology 1
```
Comment (FlyingDonkeys) [inline docs/UserGuide.md:158]: The User Guide on Google Docs uses the ```add-course``` command though? Not sure if this was a mistake or the command was changed?

**Review by FlyingDonkeys — APPROVED** [review]
Generally LGTM but there are some points that require a bit of clarification in my opinion.

**Pull request merged**